### DISCUSSION TITLE
Spotify: browse albums of the currently playing artist

### DISCUSSION THREAD

**Discussion by andyatyeo** [BimmerGestalt/AAIdrive#643] [Ideas]
Firstly, amazing job on AAIdrive to everyone that is involved in this.
This is a cheeky feature request that I think would be icing on the cake. Just putting it out there.

**The feature request:**
The feature request is to enable finding albums of an artist easier - that when you are playing spotify, that there is a menu item "Browse albums" that lists albums by the currently playing artist. (so you can select one off the list to play it).

**Why would this be useful?**
Apparently young people have moved past albums and only think of tracks. But personally I love a good album, a collection of songs that the artist has put together to make a work that is more than the sum of its parts. (and spotify tends to push into listening to the same "top 10 most popular songs of an artist" unless you can select an album to listen to)
Of course in a car, music selection has to be as streamlined as possible to limit distractions, but I feel this would be a good balance for accessing a wide range of music with minimal interaction.

**Is it even possible?**
There are technical constraints about what is/isnt possible. My feeling is that since there is some spotify webapi functionality in there already, and the spotify rest api has GET /artists/id/albums - that if the id of the currently playing artist is available then the search could be performed. 
There are already custom menu items exposed by AAIdrive (though not sure if there is a limit as to the amount of menu items).

If getting albums of an artist (GET artists/id/albums) were possible, then the same principle might also work with "get artists related artists" - GET artists/id/related-artists). Being able to pick a related artist (to the one currently playing), and then select one of their albums would be really powerful.

**Comment by hufman** [comment]
That's a great idea! The SpotifyAppController can easily make up new Browse entries, and #362 had a similar idea previously.
I think the Web API might not even be needed, I think the app connection might be enough.
